### PR TITLE
Add event categories

### PR DESCRIPTION
**Pull request by m-linner-ericsson**
### Applicable Issues
Solves AP from TC see action list [from 10 of March](https://github.com/eiffel-community/community/blob/master/meetings/MEETINGS_TC_2022.md#March-10-2022)

### Description of the Change

Groups events into categories that can be used reasoning about them or to color them when drawing them in our examples

### Alternate Designs

Selection of colors and groups should be discussed in this PR

### Benefits

We have defined colors for our examples and it makes it easier for people reasoning about them.

### Possible Drawbacks

None that I can think of

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Mattias Linnér mattias.linner@ericsson.com

### CODE REVIEW

**Comment by e-backmark-ericsson** [comment]
I believe it would be good to give each of these categories a dedicated color as well, to be used in our event graphs all over the protocol repo. For example here: https://github.com/eiffel-community/eiffel/blob/master/usage-examples/pipeline-monitoring.md#event-graph

**Comment by m-linner-ericsson** [comment]
@e-backmark-ericsson Any opinions on the colors?

**Comment by e-backmark-ericsson** [comment]
Do we think that we should recommend/propose these categories to be used as the event "family" in the routing key syntax described here: https://eiffel-community.github.io/eiffel-sepia/rabbitmq-message-broker.html ? If so, I believe we should make sure this categorization is as stable as possible over time and thereby spend some conscious thoughts on this PR before accepting it, as some routing/federation implementations might depend on those families. If not, it could be ok to let this PR through without too deep discussions.

**Review by magnusbaeck — COMMENTED** [review]
Emil put it well. If we just plan on using this in documentation then it's fine as it is, otherwise we need to be quite careful.

File: tox.ini [hunk 1]
```diff
@@ -28,7 +28,7 @@ basepython = python3
 
 [testenv:black]
 deps =
-     black==21.12b0
+     black==22.3.0
```
[the remark this answers is not in the document]
Comment (m-linner-ericsson) [reply tox.ini:31]: Yes, I probably need to rebase this change before merging so I'll solve it there.

File: eiffel-syntax-and-usage/event-categories.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+| Notification |  🟠   |
+| Source       |  🟢   |
+| Test         |  🟡   |
+| Verdict      |  🟤   |
+
+The events and their categories.
+
+| Category     | Name                                                                                                                                                                           | Abbreviation | Color |
+|:-------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------:|:-----:|
+| Activity     | [EiffelActivityCanceledEvent](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelActivityCanceledEvent.md)                                         |     ActC     |  🔵   |
```
[the remark this answers is not in the document]
Comment (m-linner-ericsson) [reply eiffel-syntax-and-usage/event-categories.md]: Fixed in [7004091](https://github.com/eiffel-community/eiffel/pull/297/commits/700409125e7e6c2bc7e247d459a5a9afffa8f9b6)

**Comment by m-linner-ericsson** [comment]
> Do we think that we should recommend/propose these categories to be used as the event "family" in the routing key syntax described here: https://eiffel-community.github.io/eiffel-sepia/rabbitmq-message-broker.html ? If so, I believe we should make sure this categorization is as stable as possible over time and thereby spend some conscious thoughts on this PR before accepting it, as some routing/federation implementations might depend on those families. If not, it could be ok to let this PR through without too deep discussions.

Agree with you both @e-backmark-ericsson and @magnusbaeck but do you have any opinions on which path to take? Is there any value of not using it for the routing key?

**Review by magnusbaeck — COMMENTED** [review]
Argh, my comment two days ago was a PR-level comment and not a Review™ so it didn't include  my draft inline comment.

**Review by t-persson — DISMISSED** [review]

**Review by e-backmark-ericsson — APPROVED** [review]
I'm ok with approving this as is now, with the addition that it should be brought up on the Eiffel Summit in a couple of weeks. There we should collect any thoughts from that audience on if further adjustments need to be done to it before recommending it to be used as the family field in the routing key through the Sepia architecture docs.

**Comment by magnusbaeck** [comment]
Let's merge this, but please note that there's a conflict to resolve first. Also, perhaps align the tox.ini to what's currently on master while you're at it?

**Review by magnusbaeck — APPROVED** [review]

**Review by e-backmark-ericsson — APPROVED** [review]

**Pull request merged**